dash-pipeline/bmv2/dash_pipeline.p4

```diff
@@ -83,18 +83,50 @@ control dash_ingress(inout headers_t hdr,
         }
     }
 
+#define ACL_GROUPS_PARAM(prefix) \
+    bit<16> ## prefix ##_stage1_dash_acl_group_id, \
+    bit<16> ## prefix ##_stage2_dash_acl_group_id, \
+    bit<16> ## prefix ##_stage3_dash_acl_group_id, \
+    bit<16> ## prefix ##_stage4_dash_acl_group_id, \
+    bit<16> ## prefix ##_stage5_dash_acl_group_id
+
+#define ACL_GROUPS_COPY_TO_META(prefix) \
+   meta.stage1_dash_acl_group_id = ## prefix ##_stage1_dash_acl_group_id; \
+   meta.stage2_dash_acl_group_id = ## prefix ##_stage2_dash_acl_group_id; \
+   meta.stage3_dash_acl_group_id = ## prefix ##_stage3_dash_acl_group_id; \
+   meta.stage4_dash_acl_group_id = ## prefix ##_stage4_dash_acl_group_id; \
+   meta.stage5_dash_acl_group_id = ## prefix ##_stage5_dash_acl_group_id;
+
     action set_eni_attrs(bit<32> cps,
                          bit<32> pps,
                          bit<32> flows,
                          bit<1> admin_state,
                          IPv4Address vm_underlay_dip,
-                         bit<24> vm_vni) {
+                         bit<24> vm_vni,
+                         ACL_GROUPS_PARAM(inbound_v4),
```

Comment:
There should also be a differentiation between inbound and outbound group IDs. Inbound v4 ACL stage 1 != Outbound v4 ACL stage 1.